### ISSUE TITLE
2.1.14 - Barcode Entry Form not processing

### ISSUE THREAD

**Issue by LoganGoesPlaces** [geoffhumphrey/brewcompetitiononlineentry#1027]
Version: 
2.1.14

Installation URL: 
https://unyha.com/comp

Is your installation hosted on brewcompetition.com or brewcomp.com?
No

Description of Issue:
We changed to a new hosting provider this winter and this is our first time trying to use BCOE&M with the new hosting.  We hadn't noticed any issues during setup but we are now testing everything before we officially open up our competition and we noticed that the barcode entry check-in page is not processing our submissions.  When we submit it reloads the page but all that loads is the menu bar and the "Administration: Entry Check-in" title.  When we check the entries list none of the tested entry numbers have the Rec'd checkbox toggled.  I can't find any errors and I thought maybe increasing the connection timeout and php memory limit might affect the issue but no luck.

For enhancements, please prepend "Enhancement - " to the title. For issues/bugs, please prepend the BCOE&M version number the title.

**Comment by geoffhumphrey** [comment]
This was an issue with the 2.1.14 release. If you update the site to the latest version, this issue should be resolved. Using that version, we successfully checked in 150+ entries into the database for the last comp I organized. If updating to the latest version doesn't work, please post back here.

**Comment by LoganGoesPlaces** [comment]
Geoff,

Thank you for the quick response.  I'll give that a try and report back.

Logan

**Comment by LoganGoesPlaces** [comment]
I believe it is working after the update.  Sorry, I looked around through the change notes and hadn't noticed the issue mentioned.

**Comment by LoganGoesPlaces** [comment]
Unfortunately It would appear I was premature in closing this issue.  Upon further testing we are still experiencing the problem.

After upgrade I WAS able to submit one entry.  But the next time we tried testing the form it was back to not working at all.  I think this is the same thing that has been happening for us and I wonder if maybe the first entry doesn't finish processing completely or something along those lines.

The very first time we tested it the first entry worked and none of the following entries processed.  Then when I was troubleshooting the issue I was able to get one entry to submit then none of the following.

To troubleshoot I've tried clearing cache & cookies, wiping our server's cache, and testing on both Chrome and Firefox.  

When this was happening before I didn't see anything in our error log but I just came across this:

> Got error 'PHP message: PHP Fatal error:  Uncaught Error: [] operator not supported for strings in .../includes/process/process_barcode_check_in.inc.php:28\nStack trace:\n#0 .../admin/barcode_check-in.admin.php(40): include()\n#1 .../index.php(184): include('...')\n#2 {main}\n  thrown in .../includes/process/process_barcode_check_in.inc.php on line 28\n'

**Comment by LoganGoesPlaces** [comment]
The line generating the error is adding the current judging & entry number to $flag_jnum when $row_jnum count is above zero.

I did a little testing by just submitting two of our entry numbers with no other fields filled out.  The script was failing while processing the first entry.  $row_jnum count was returning 1 but I was unable to output anything from it.

Instead of the shorthand array push I tried using array_push() and the submissions seem to be working now.  I'd expect the existing shorthand to work, though.  Should I be seeing random judging numbers being generated?  All of my tests assign the judging number 000000 (all of the entries I tested are < 9) unless I enter something.  

I'm not sure what else may be affected by that array being inaccurate - this is first time using BCOE&M so trying to get my bearings.
